src/solidlsp/language_servers/intelephense.py

```diff
@@ -79,17 +79,23 @@ def _setup_runtime_dependencies(cls, logger: LanguageServerLogger, config: Langu
 
         return f"{intelephense_executable_path} --stdio"
 
-    def __init__(self, config: LanguageServerConfig, logger: LanguageServerLogger, repository_root_path: str):
+    def __init__(
+        self,
+        config: LanguageServerConfig,
+        logger: LanguageServerLogger,
+        repository_root_path: str,
+        intelephense_options: dict[str, int] | None = None,
```

Comment:
see above. We already have LanguageServerConfig - this is where such parameters should be naturally set, I think.

Language servers that don't support max memory and max file should log a warning that this option is ignored

 @opcode81 wdyt?